server/src/main/java/io/seata/server/session/GlobalSession.java

```diff
@@ -122,6 +122,14 @@ public boolean isTimeout() {
         return (System.currentTimeMillis() - beginTime) > timeout;
     }
 
+    /**
+     * prevent could not handle rollbacking transaction
+     * @return if true force roll back
+     */
+    public boolean isRollbackingDead() {
+        return (System.currentTimeMillis() - beginTime) > (3 * 6000);
```

Comment:
(System.currentTimeMillis() - beginTime) > (3 * 6000 + timeout);
Suggested overload isTimeout()

Reply:
My mistake, you are right!